### ISSUE TITLE
What is a workflow to update Backend environments when Web Previews is enabled?

### ISSUE THREAD

**Issue by Nicktho** [aws-amplify/amplify-console#2402]
**Please describe which feature you have a question about?**

When Web previews is enabled on a branch and the option to create a new backend environment for each branch is enabled, what is the workflow to update backend environments? It is not clear to me from the documentation if this new backend environment is suitable to update, and if so, the workflow to do so.

**Provide additional details**

For example, an amplify project is deployed to github on the `main` branch, which automatically deploys a frontend environment (`main`) and a backend environment (`prod`) in amplify on every commit.

A developer wants to add a feature to the backend to the project, they create a new branch.

`git checkout -b new-feature` 

From here, if the developer wants to add an amplify backend feature, the only environment they have locally is `prod` (reflecting the environment of the `main` branch), as the pull request and subsequent ephemeral backend environment has not been created yet. Given that, they enter the following:

`amplify add api`

Imagine the developer goes through the interactive prompts and chooses the following options: 

![image](https://user-images.githubusercontent.com/8622243/142977256-4919baf2-6f7e-41ff-b0a5-9dfbd44c05a2.png)

The developer does not `amplify push` here, as they do not want to update the production environment, so they commit their changes and push it to github.

`git add .`
`git commit -m 'add api'`
`git push origin new-feature`

Afterwards, they open a PR on github, which Amplify Previews notices and starts provisioning a frontend and backend environment.

The backend environment that Amplify provisions includes the new api changes that were made, and let's say was named `prb`. 

Through practice, here I know that some files are needed to be pulled from the environment and committed to the repository.

The developer then wants to `amplify env checkout prb` to `amplify pull` those changes, but the environment does not exist and is met with an error:

![image](https://user-images.githubusercontent.com/8622243/142977641-e28f37af-db23-40d0-8386-67c302031a62.png)

Running `amplify env list` does not list the new backend environment, `prb`
![image](https://user-images.githubusercontent.com/8622243/142977680-1c9e9510-c530-4b0e-9af4-ee8efc6d0dbb.png)

---
**Side note**: , here I tried `amplify env pull` and was met with unexpected results.  I expected `amplify env pull` to make no local changes, as the `prod` environment has not been pushed, but the results are the following:

![image](https://user-images.githubusercontent.com/8622243/142977780-27895e71-c70d-4564-a693-57a86b618041.png)

![image](https://user-images.githubusercontent.com/8622243/142977826-9f0ea095-ca30-4d1d-a61a-65c19906a6b4.png)

![image](https://user-images.githubusercontent.com/8622243/142977849-ddf18d55-1bda-4492-867f-57524e0bfb73.png)

Here, it appears that `amplify env pull` in the `prod` environment pulls down something that refers to a graphQL api, even though nothing was pushed. Strangely, running `amplify env pull` in a fresh clone of the repository, after an initial `amplify init` to set it up, did not produce these results, why did the previous `amplify env pull` pull a change to the previous file?

![image](https://user-images.githubusercontent.com/8622243/142978006-71ec1e5f-aa43-4508-982b-bf412c5f82c8.png)
---

Continuing on, the developer ends up browsing the Amplify Console to find the backend environment that was created, `prb`, and uses the `amplify pull --appId APP_ID --envName prb` command to force their local environment to recognise the ephemeral environment `prb`:

![image](https://user-images.githubusercontent.com/8622243/142978362-f77b6195-79fd-4f6d-8cc0-104a888438ee.png)

The developer can now commit the code changes that were produced by the Previews deployment back to the branch:

![image](https://user-images.githubusercontent.com/8622243/142978428-dbf48ae6-7264-4602-8c7b-cc3b25b4cd6b.png)

Unfortunately, this includes adding the `prb` environment to `amplify/team-provider-info.json`, which the developer knows will no longer exist once the pull request is merged and the branch deleted, how would the developer avoid this? Should they not commit the change to `amplify/team-provider-info.json`, or does a new commit need to be made to `main` after the PR is merged, removing the environment from the file.

Is this the intended workflow when working with web previews?

Should the developer instead work in a sandboxed backend environment locally, so that they can `amplify push` and get the code changes required in before opening the PR? Meaning that two backend environments is created per PR (the sandbox, and the automatically created one by Previews).

**What AWS Services are you utilizing?**
Amplify

**Provide additional details e.g. code snippets**
In summary:
* What is the recommended workflow for a team when developing backend environments while Previews is turned on for a github repository?
* How can a developer pull changes made by Previews provisioning changes to a new backend environment back into the codebase?
* Why does `amplify env pull` modify local files after an `amplify add api` and `git push` to a Previews enabled repository, without `amplify push`-ing any changes to the original environment the `amplify env pull` was made in? (And why do we not see this on a fresh clone of a repository).

**Comment by Nicktho** [comment]
The `App ID` is `d14291rfs1uwyz` and region is `ap-southeast-2`

**Comment by andreroggeri** [comment]
So sad to see an incredible effort to explain a basic workflow and get no answer from AWS.

And everyone that comes here hoping to solve the problem and leaves wondering what Nick did to solve that 😞
